libbeat/outputs/amqp/client.go

```diff
@@ -467,6 +472,11 @@ func (c *client) prepareEvent(codec codec.Codec, incoming eventTracker, now time
 	}
 	c.logger.Debugf("calculated routing key: %v", routingKey)
 
+	headers, err := c.getHeaders(content)
+	if err != nil {
+		return nil, err
```

Comment:
🍹 adding a bit of message context at each step helps to maintain error detail when automatic traces aren't available. It's a good habit to get into for Go projects, but isn't 100% necessary here (hence the 🍹 ). If you want, you can have a look at the error messages you're returning from `getHeaders` and see if there's any commonality that you can move to here instead (like the above).